lib/database.js

```diff
@@ -41,6 +41,13 @@ exports.DatabaseAPI = function DatabaseAPI(hoodie, options) {
 
   db.add = function (type, attrs, callback) {
     attrs.type = type;
+
+    // we can prepare a couch `id` when no `doc.id` is supplied, so
+    // let's check whether it exists first...
+    if (!attrs.id) {
+      return callback(new Error('No `id` associated with this Hoodie document'));
```

Comment:
we call em “objects” in Hoodie, to differentiate from CouchDB “documents” (which represent Hoodie objects, but look differently